# Assignment-6

## ESTIMATION AND CONFIDENCE INTERVALS

#### Background
In quality control processes, especially when dealing with high-value items, destructive sampling is a necessary but costly method to ensure product quality. The test to determine whether an item meets the quality standards destroys the item, leading to the requirement of small sample sizes due to cost constraints.


#### Scenario
A manufacturer of print-heads for personal computers is interested in estimating the mean durability of their print-heads in terms of the number of characters printed before failure. To assess this, the manufacturer conducts a study on a small sample of print-heads due to the destructive nature of the testing process

### Data
>A total of 15 print-heads were randomly selected and tested until failure. The durability of each print-head (in millions of characters) was recorded as follows:
1.13, 1.55, 1.43, 0.92, 1.25, 1.36, 1.32, 0.85, 1.07, 1.48, 1.20, 1.33, 1.18, 1.22, 1.29


In [1]:
import pandas as pd
from scipy import stats 
import numpy as np

In [2]:
num=[1.13, 1.55, 1.43, 0.92, 1.25, 1.36, 1.32, 0.85, 1.07, 1.48, 1.20, 1.33, 1.18, 1.22, 1.29]
data=pd.DataFrame(num,columns=['A'])
data.head()

,A
0,1.13
1,1.55
2,1.43
3,0.92
4,1.25


#### a. Build 99% Confidence Interval Using Sample Standard Deviation
Assuming the sample is representative of the population, construct a 99% confidence interval for the mean number of characters printed before the print-head fails using the sample standard deviation. Explain the steps you take and the rationale behind using the t-distribution for this task.


In [3]:
data.shape

(15, 1)

In [4]:
std=data.std()
std

A    0.193164
dtype: float64

In [5]:
mean_a=data.mean()

In [6]:
#T-Test
cint=stats.t.interval(0.99,df=14,loc=mean_a,scale=std/np.sqrt(15))
np.round(cint,3)

array([[1.09 ],
       [1.387]])

##### b. Build 99% Confidence Interval Using Known Population Standard Deviation
If it were known that the population standard deviation is 0.2 million characters, construct a 99% confidence interval for the mean number of characters printed before failure.


In [7]:
#Z-Test
cint=stats.norm.interval(0.99,loc=mean_a,scale=0.2/np.sqrt(15))
np.round(cint,3)

array([[1.106],
       [1.372]])

## Hypothesis Test

##### Background:

> Bombay hospitality Ltd. operates a franchise model for producing exotic Norwegian dinners throughout New England. The operating cost for a franchise in a week (W) is given by the equation W = $1,000 + $5X, where X represents the number of units produced in a week. Recent feedback from restaurant owners suggests that this cost model may no longer be accurate, as their observed weekly operating costs are higher.


##### Objective:
> To investigate the restaurant owners' claim about the increase in weekly operating costs using hypothesis testing.


##### Data Provided:
-	The theoretical weekly operating cost model: W = $1,000 + $5X
-	Sample of 25 restaurants with a mean weekly cost of Rs. 3,050
-	Number of units produced in a week (X) follows a normal distribution with a mean (μ) of 600 units and a standard deviation (σ) of 25 units


### Assignment Tasks:
#### 1. State the Hypotheses statement:


In [8]:
import numpy as np

# t-tests
import scipy.stats as stats
from scipy.stats import ttest_1samp, ttest_ind, ttest_rel

# z-tests
import statsmodels.api as sm

##### Null Hypothesis (H₀)

The weekly operating cost has not increased and follows the theoretical model.

                          H0:u=4000

##### Alternative Hypothesis (H₁)

The weekly operating cost is higher than the theoretical model.

                          𝐻1:u>4000


#### 2. Calculate the Test Statistic:
Use the following formula to calculate the test statistic(z):
##### where:
 -	ˉxˉ = sample mean weekly cost (Rs. 3,050)
 -	μ = theoretical mean weekly cost according to the cost model (W = $1,000 + $5X for X = 600 units)
 -	σ = 5*25 units
 -	n = sample size (25 restaurants)


In [22]:
#Population Mean (u):
population_mean = 4000

#Population Standard Deviation:
population_standard_deviation = 125

#Significance Level (5%)
alpha  = 0.05 

#sample size (n)
sample_size  = 25

#sample mean (x) 
sample_mean  = 3050

In [27]:
#z_score Formula
z_score = (sample_mean - population_mean) / (population_standard_deviation / np.sqrt(25))
#p_value
p_value = 1 - stats.norm.cdf(z_score) 
p_value

np.float64(1.0)

In [24]:
z_critical = stats.norm.ppf(1 - alpha)

In [25]:
print("Z-score:", z_score)
print("P-value:", p_value)
print("Critical value:", z_critical)

Z-score: -38.0
P-value: 1.0
Critical value: 1.6448536269514722


#### 3. Determine the Probability and compare:
Using the alpha level of 5% (α = 0.05),


In [21]:
if z_score > z_critical:
    print("Reject the null hypothesis")
else:
    print("Fail to reject the null hypothesis")

Fail to reject the null hypothesis


#### 4. Make a Decision: Compare the test statistic with the critical value to decide whether to reject the null hypothesis.

Because the test statistic (-38) is less than the critical value (1.645) and the p-value is greater than 0.05, we fail to reject the null hypothesis.


#### Conclusion:

There is no strong statistical evidence to support the restaurant owners' claim that the weekly operating costs are higher than predicted by the theoretical cost model.

In fact, the sample mean weekly cost (Rs. 3050) is lower than the expected theoretical cost (Rs. 4000).